src/i18n.ts

```diff
@@ -0,0 +1,342 @@
+import has from 'dojo-core/has';
+import global from 'dojo-core/global';
+import { Handle } from 'dojo-core/interfaces';
+import { assign } from 'dojo-core/lang';
+import Promise from 'dojo-shim/Promise';
+
+/**
+ * A default bundle used as basis for loading locale-specific bundles.
+ */
+export interface Bundle<T extends Messages> {
+	/**
+	 * The absolute module ID used by the loader to resolve paths to locale-specific paths. This MUST follow the format
+	 * "{basePath}{separator}{filename}". For example, if the module ID for a bundle is "nls/common", the loader will
+	 * look for locale-specific bundles at "nls/{locale}/common".
+	 */
+	baseUrl: string;
```

Comment:
No, especially if you want to indicate to someone they can't modify the bundle.  I forgot to check, do you have `strictNullChecks` and `noImplicitThis`?  The only problem you can run into is that if you "build up" the creation of the object, but you can work around by casting to this interface on the function return.